18F/identity-idp
# LG-6314: Add Start Over and Cancel links to password confirmation

**Why**: For feature parity with the existing screen, a user should have the option to start over or cancel the proofing process.

**Testing Instructions:**

You will need to set the step as enabled in your local `config/application.yml`:

```yml
development:
  idv_api_enabled_steps: '["password_confirm","personal_key","personal_key_confirm"]'
```

1. Navigate to http://localhost:3000
2. Sign in
3. Navigate to http://localhost:3000/verify
4. Complete proofing flow up to password confirmation step
5. Observe "Start Over" and "Cancel" links at the bottom of the page
6. Click "Cancel"
7. Observe you are navigated to the cancel confirmation screen, including `?step=password_confirm` in the URL
8. Click "Go back"
9. Observe you are returned to the password confirmation screen
10. Click "Start over"
11. Observe you are returned to the start of the proofing process

**Implementation Notes:**

This includes quite a bit of shuffling of components already implemented in document capture, so that they can be reused in both places.

This also implements a new "flow context", whose purpose overlaps with the one being introduced by @nprimak in #6323.

**Screenshots:**

Before|After
---|---
![localhost_3000_verify_v2_password_confirm](https://user-images.githubusercontent.com/1779930/168344119-23fe5ed4-c505-4815-a56f-19a540c1abe8.png)|![localhost_3000_verify_v2](https://user-images.githubusercontent.com/1779930/168344121-b9b52d90-165c-42cf-873b-09e8a7251575.png)
